### PR TITLE
Adds ability to pass in map of file format options when creating a PailSpec.

### PR DESCRIPTION
**Pull request by dacamo76**
By default the Pail file format is Hadoop SequenceFile.
There are three SequenceFileWriters each based on a compression type (none, record, block). The compression codec may also be specified.

This change allows us to pass compression type and compression codec options for Pail to use when creating the underlying SequenceFile.

### CODE REVIEW

File: src/main/clojure/clj_pail/core.clj [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 
 
 (defn ^PailSpec spec
   "Builds a PailSpec from a PailStructure."
-  [^PailStructure structure]
-  (PailSpec. structure))
+  ([^PailStructure structure]
+     (PailSpec. structure))
+  ([^PailStructure structure opts]
+     (PailSpec. PailFormatFactory/SEQUENCE_FILE opts structure)))
```
Comment (dcuddeback) [inline src/main/clojure/clj_pail/core.clj]: If I'm reading this correctly, this is going to use the following constructor:

``` java
public PailSpec(String name, Map<String, Object> args, PailStructure structure)
```

Which means the `PailSpec` will have a name of `"SequenceFile"`. Am I reading that correctly?

Comment (dacamo76) [inline src/main/clojure/clj_pail/core.clj]: Correct.

Comment (dacamo76) [inline src/main/clojure/clj_pail/core.clj]: Not sure if you're doubting the explicit argument `PailFormatFactory/SEQUENCE_FILE` in the call to `PailSpec`. I'm not a fan of explicitly setting arguments inside a function, but I think it's better than forcing a user to pass in a name to the `spec` function. My thinking is:
- This is the way `PailSpec` does it. If you don't pass in a name, it defaults to `"SequenceFile"`. Those familiar with Pail will be used to this behavior.
- At the moment, `"SequenceFile"` is the only pre-defined format.
- Forcing user to pass a name into `spec` adds complexity without much gain, most users will probably end up partially applying `spec` every time they use the library. 
- Passing in a name to `PailSpec` is the only way to get `PailSpec` to recognize the options. (commit message 862ab2c expands on this a little)
- The name must be `"SequenceFile"` else `PailSpec` ignores the options completely (nathanmarz/dfs-datastores#33). A name other than `"SequenceFile"` implies a custom implementation of `PailFormat`.

In my opinion, the correct way for `PailSpec` to handle this is to check for name and options parameters explicitly, if the name is empty and there are options present, use the default name `"SequenceFile"` and copy the options into the `PailSpec`. Unfortunately, Pail seems to be in sleep mode until the release of the Big Data book.

Comment (dcuddeback) [inline src/main/clojure/clj_pail/core.clj]: @dacamo76 Thanks for explaining your thought process on this. That makes a lot more sense now. I was confused by the purpose of a PailSpec's name. I'll take another look at this soon.

**Comment by dacamo76** [comment]
@dcuddeback After taking another look at this, I've changed course. We should force a user to pass in a "name" when using the options parameter. By implicitly setting the name to "SequenceFile" we are dangerously relying on implementations details.

The PailFormat is the parameter called "name" in the PailSpec constructors. The "options" are tied specifically to the "name" parameter. When you create a Pail, internally PailFormatFactory.create(spec) is called which instantiates a new PailFormat according to the "name" parameter in the spec, passing in the spec options as constructor args.

In other words, the PailSpec "name" parameter is used to instantiate a PailFormat of that name with the "options" parameter as its constructor arg.

**Comment by dcuddeback** [comment]
@dacamo76 That makes sense to me, especially after reading `PailFormatFactory.create(spec)`.
